### ISSUE TITLE
Using monoforest to reduce the size of LightGBM/PMML models?

### ISSUE THREAD

**Issue by thvasilo** [catboost/catboost#1986]
Hello, in the Monoforest paper (end of Section 4) it is mentioned that it's possible to take a model trained in LightGBM and transform it to "to an ensemble of symmetric trees", which I assume are oblivious trees.

I was going through the [Monoforest tutorial](https://github.com/catboost/tutorials/blob/master/model_analysis/monoforest_tutorial.ipynb) but it's not clear how we can accomplish that using the functions available in the [monoforest module](https://github.com/catboost/catboost/blob/master/catboost/python-package/catboost/monoforest.py). 

Is there a way to convert a LightGBM or PMML-formatted ensemble to an oblivious one (Algorithm 1 in the paper) using the C++ or Python code available in the open-source package or was this only implemented for the paper?

**Comment by ek-ak** [comment]
Hello! You can convert LightGBM model into ONNX and then load ONNX model `model = catbost.CatBoost().load_model(fname, format='onnx') `and use monoforest module.

**Comment by thvasilo** [comment]
Thanks! And after creating the polynomials is there a way to convert to an oblivious ensemble using Python?

I can see there's a [C++ API](https://github.com/catboost/catboost/blob/master/catboost/libs/monoforest/polynom.cpp#L233), but not sure if it's exposed to Python.

My understanding is that I'll need to use C++ to convert the polynomials to an oblivious ensemble.

**Comment by kizill** [comment]
No, currently there is no such binding, but you can easily add them here in Cython code: https://github.com/catboost/catboost/blob/master/catboost/python-package/catboost/_monoforest.pxi#L118
You can write some c++ helper functions in https://github.com/catboost/catboost/blob/master/catboost/python-package/catboost/monoforest_helpers.h and corresponding .cpp file.
If you'll need any help with c++ code - feel free to find me in our official Telegram chat [@catboost_en](https://t.me/catboost_en)
